apptests/kind/kind.go

```diff
@@ -0,0 +1,78 @@
+// Package kind provides function for creating and managing kind clusters.
+package kind
+
+import (
+	"context"
+	"os"
+
+	"sigs.k8s.io/kind/pkg/cluster"
+	"sigs.k8s.io/kind/pkg/cmd"
+)
+
+type Cluster struct {
+	provider           *cluster.Provider
+	kubeconfigFilePath string
+	name               string
+}
+
+const defaultClusterName = "kind"
+
+// CreateCluster creates a new kind cluster with the given name.
+func CreateCluster(ctx context.Context, name string) (*Cluster, error) {
```

Comment:
Nit: Can we split the instance creation of the cluster and the actual cluster creation as we did on K-E2E. This has the advantage of validating the inputs before creating the cluster. It also has a more consistent set of methods (the object has the delete but not the create).